compiler/passes/normalize.cpp

```diff
@@ -2683,6 +2745,70 @@ static bool skipFixup(ArgSymbol* formal, Expr* domExpr, Expr* eltExpr) {
   return false;
 }
 
+static bool isArrayFormal(ArgSymbol* arg) {
+  if (BlockStmt* typeExpr = arg->typeExpr) {
+    //
+    // The body is usually a single callExpr.  However there are rare
+    // cases in which normalization generates one or more call_temps
+    // i.e. a sequence of defExpr/primMove pairs.
+    //
+    // In either case the desired callExpr is the tail of the body.
+    //
+
+    if (CallExpr* call = toCallExpr(typeExpr->body.tail)) {
+      if (call->isNamed("chpl__buildArrayRuntimeType")) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+static void fixupExportedArrayFormals(FnSymbol* fn) {
+  CallExpr* retCall = toCallExpr(fn->body->body.tail);
+  INT_ASSERT(retCall);
+  CallExpr* callOrigFn = retCall;
+  if (retCall->isPrimitive(PRIM_RETURN)) {
+    INT_ASSERT(retCall->numActuals() == 1);
+    callOrigFn = toCallExpr(retCall->get(1));
+  }
+
+  INT_ASSERT(callOrigFn);
+
+  for_formals(formal, fn) {
+    if (isArrayFormal(formal)) {
+      BlockStmt*            typeExpr = formal->typeExpr;
+
+      CallExpr*             call     = toCallExpr(typeExpr->body.tail);
+      int                   nArgs    = call->numActuals();
+      Expr*                 eltExpr  = nArgs == 2 ? call->get(2) : NULL;
```

Comment:
What is this call we might be working with? Why is argument 2 important? Needs at least some comments here about what it's trying to match.